src/game_events/action_wml.cpp

```diff
@@ -894,8 +894,25 @@ WML_HANDLER_FUNCTION(unit,, cfg)
 		.allow_rename_side(true)
 		.allow_show(true);
 
-	uc.add_unit(parsed_cfg, &cfg);
+	try
+	{
+		uc.add_unit(parsed_cfg, &cfg);
+	}
+	catch(const unit_type::error& e)
+	{
+		std::string unit_type_s = parsed_cfg["parent_type"].blank() ? parsed_cfg["type"].str() : parsed_cfg["parent_type"].str();
 
+		if(unit_type_s.empty()) {
+			ERR_WML << "Empty unit type inside [unit] WML";
+		} else {
+			const unit_type* i = unit_types.find(unit_type_s);
+			if(!i) {
+				ERR_WML << "Unkown unit type inside [unit] WML: " << unit_type_s;
+			} else {
+				ERR_WML << "Unkown error occured inside [unit] WML: " << e.what();
```

Comment:
I'd just have a single error message and rely on `what()` to contain the reason instead of duplicating logic and trying to figure it out again. At a minimum calling this an unknown error here is just confusing.

Reply:
I did it this way to have better looking error messages. The unkown error shouldnt really occur it is there just in case

Reply:
If the exception message is not good looking enough then that should be improved. Duplicating logic (that will just diverge with time) to guess what the exception is about is not good.

Reply:
Simplified the error message and the exception is rethrown.